test/box-luatest/buffer_test.lua

```diff
@@ -0,0 +1,86 @@
+local buffer = require('buffer')
+local ffi = require('ffi')
+local tarantool = require('tarantool')
+local t = require('luatest')
+
+ffi.cdef[[
+int
+__asan_address_is_poisoned(void const volatile *addr);
+]]
+
+local g = t.group()
+
+g.test_poison = function()
```

Comment:
You are using Tarantool-runner in this test, which can lead to various problems.

Our visualisation of test framework looks like:
```
Test-run(py vm) runs:
   1 Tarantool(luatest) runs:  <-- if something is changed here, then it can affect luatest framework
       1 tarantool (luatest.Server); <-- isolated from all
       2 tarantool (luatest.Server);
       3 tarantool (luatest.Server);
       ...
   2 Tarantool(luatest) runs:
       1 tarantool (luatest.Server);
       2 tarantool (luatest.Server);
       ...
   ...
```

If it's not difficult for you, transfer all tests to the `luatest.Server`:

```lua
g.before_all(function()
   g.s = Server:new()

g.test_poison = function()
    t.skip_if(not tarantool.build.asan, 'only make sense with ASAN')
    g.s:exec(function()
         ...
    end)
end

g.after_all(function()
   g.s:drop()
```

This server isolation helps to achieve test stability + it is convenient to look at "what went wrong" in the server artifacts.

Reply:
Done.